modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -1400,6 +1400,125 @@ struct PowerFunctor : public BaseFunctor
     int64 getFLOPSPerElement() const { return power == 1 ? 2 : 10; }
 };
 
+struct ExpFunctor : public BaseFunctor
+{
+    typedef ExpLayer Layer;
+    float base, scale, shift;
+    float normScale, normShift;
+
+    ExpFunctor(float base_ = -1.f, float scale_ = 1.f, float shift_ = 0.f)
+        : base(base_), scale(scale_), shift(shift_)
+    {
+        CV_Assert(base == -1.f || base > 0.f);
```

Comment:
Please use CV_Check*() macros (instead of CV_Assert) to emit better error messages:

```
CV_Check(base, base == -1.f || base > 0.f, "Unsupported 'base' value");
```

Reply:
Fixed.